tests/test_mpi.py

```diff
@@ -1005,6 +1008,182 @@ def test_avoid_haloupdate_if_distr_but_sequential(self, mode):
+        pde_v = v.dt - (tau.dx)
+        pde_tau = tau.dt - ((v.forward).dx)
+        u_v = Eq(v.forward, solve(pde_v, v.forward))
+
+        u_tau = Eq(tau.forward, solve(pde_tau, tau.forward))
+
+        # Test two variants of receiver interpolation
+        nrec = 1
+        rec = SparseTimeFunction(name="rec", grid=grid, npoint=nrec, nt=tn)
+        rec.coordinates.data[:, 0] = np.linspace(0., shape[0], num=nrec)
+
+        # The receiver 0
+        rec_term0 = rec.interpolate(expr=v)
+
+        # The receiver 1
+        rec_term1 = rec.interpolate(expr=v.forward)
+
+        # Test receiver interpolation 0, here we have a halo exchange hoisted
+        op0 = Operator([u_v] + [u_tau] + rec_term0)
+
+        calls = [i for i in FindNodes(Call).visit(op0)
+                 if isinstance(i, HaloUpdateCall)]
+
+        # The correct we want
+        assert len(calls) == 3
+
+        assert len(FindNodes(HaloUpdateCall).visit(op0.body.body[1].body[1].body[0])) == 1
+        assert len(FindNodes(HaloUpdateCall).visit(op0.body.body[1].body[1].body[1])) == 2
+        assert calls[0].arguments[0] is v
+        assert calls[1].arguments[0] is tau
+        assert calls[2].arguments[0] is v
+
+        # Test receiver interpolation 1, here we should not have any halo exchange
+        # hoisted
+        op1 = Operator([u_v] + [u_tau] + rec_term1)
+
+        calls = [i for i in FindNodes(Call).visit(op1)
+                 if isinstance(i, HaloUpdateCall)]
+
+        # The correct we want
+        assert len(calls) == 3
+
+        assert len(FindNodes(HaloUpdateCall).visit(op1.body.body[1].body[0])) == 0
+        assert len(FindNodes(HaloUpdateCall).visit(op1.body.body[1].body[1])) == 3
+        assert calls[0].arguments[0] is tau
+        assert calls[1].arguments[0] is v
+        assert calls[2].arguments[0] is v
+
+        # Further complicate/stree-test adding an artifical example
+        # with two hoisting opportunities
+
+        # Velocity and pressure fields
+        v2 = TimeFunction(name='v2', grid=grid, space_order=so)
+        tau2 = TimeFunction(name='tau2', grid=grid, space_order=so)
+
+        # First order elastic-like dependencies equations
+        pde_v2 = v2.dt - (tau2.dx)
+        pde_tau2 = tau2.dt - ((v2.forward).dx)
+        u_v2 = Eq(v2.forward, solve(pde_v2, v2.forward))
+
+        u_tau2 = Eq(tau2.forward, solve(pde_tau2, tau2.forward))
+
+        # Test two variants of receiver interpolation
+        nrec = 1
+        rec2 = SparseTimeFunction(name="rec2", grid=grid, npoint=nrec, nt=tn)
+        rec2.coordinates.data[:, 0] = np.linspace(0., shape[0], num=nrec)
+
+        # The receiver 2
+        rec_term2 = rec2.interpolate(expr=v2)
+
+        # The receiver 3
+        rec_term3 = rec2.interpolate(expr=v2.forward)
+
+        # Test receiver interpolation 0, here we have a halo exchange hoisted
+        op2 = Operator([u_v] + [u_v2] + [u_tau] + [u_tau2] + rec_term0 + rec_term2)
```

Comment:
Should this and `op3` be split off in their own tests, or is it necessary to have them in this test?

Reply:
ah, yes... I was writing the same above. This must be split over multiple tests